AppKit/CPWindow/CPWindow.j

```diff
@@ -3844,6 +4069,46 @@ var interpolate = function(fromValue, toValue, progress)
 
 @end
 
+@implementation CPWindow (TrackingArea)
+
+- (void)_addToViewsWithTrackingAreas:(CPView)aView
+{
+    [_viewsWithTrackingAreas addObject:aView];
+    
+    var tas = [aView trackingAreas];
+    
+    for (var i = 0, nb = [tas count]; i < nb; i++)
+        [self _addTrackingArea:[tas objectAtIndex:i]];
```

Comment:
Okay, now I look at the logic more closely, the whole thing can be simplified since a tracking area knows the view it belongs to. In addTrackingArea, you only need to call `[_window _addTrackingArea:trackingArea]`, and in that method, if the tracking area's view is not yet in _viewsWithTrackingAreas, add it. In _setWindow you can call _addTrackingAreaView, which stays the same.

Conceptually this way is much cleaner, because the view doesn't know anything about the implementation in CPWindow.